### PR TITLE
Enable tests (spix et cie) on OSX

### PR DESCRIPTION
**Pull request by nirvn**
Ultimately hoping this'll give us antialiased spix screensots.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 7c8494889e00c2c49221d093613a23b5a5929401: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3373-ee3c95cfa3b91485cc07efe639649990a4d386e9-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3373-ee3c95cfa3b91485cc07efe639649990a4d386e9-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3373-ee3c95cfa3b91485cc07efe639649990a4d386e9-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3373-ee3c95cfa3b91485cc07efe639649990a4d386e9-x86-android.apk)
</details>

**Comment by nirvn** [comment]
@m-kuhn , here's an interesting challenge :) when running the qfield_spix binary (as part of the smoke test), the proj.db (and QGIS's srs.db) are not found due to those files copied in a different path than the established relative path norm.

This is where the files are copied to:

```
2022-09-26T04:10:39.6760200Z /Users/runner/work/_temp/-2070376416/cmake-3.23.2-macos-universal/CMake.app/Contents/bin/cmake -E make_directory /Users/runner/builddir/output/bin/qgis/plugins
2022-09-26T04:10:39.6849750Z /Users/runner/work/_temp/-2070376416/cmake-3.23.2-macos-universal/CMake.app/Contents/bin/cmake -E make_directory /Users/runner/builddir/output/bin/qfield.app/Contents/share/qgis/resources
2022-09-26T04:10:39.6934180Z /Users/runner/work/_temp/-2070376416/cmake-3.23.2-macos-universal/CMake.app/Contents/bin/cmake -E copy_directory /Users/runner/builddir/vcpkg_installed/x64-osx/share/qgis/resources /Users/runner/builddir/output/bin/qfield.app/Contents/share/qgis/resources
2022-09-26T04:10:40.9541620Z /Users/runner/work/_temp/-2070376416/cmake-3.23.2-macos-universal/CMake.app/Contents/bin/cmake -E rm -- /Users/runner/builddir/output/bin/qfield.app/Contents/share/qgis/resources/data/world_map.gpkg
2022-09-26T04:10:40.9629640Z /Users/runner/work/_temp/-2070376416/cmake-3.23.2-macos-universal/CMake.app/Contents/bin/cmake -E make_directory /Users/runner/builddir/output/bin/qfield.app/Contents/share/proj
2022-09-26T04:10:40.9706450Z /Users/runner/work/_temp/-2070376416/cmake-3.23.2-macos-universal/CMake.app/Contents/bin/cmake -E copy_directory /Users/runner/builddir/vcpkg_installed/x64-osx/share/proj /Users/runner/builddir/output/bin/qfield.app/Contents/share/proj
2022-09-26T04:10:41.0674620Z /Users/runner/work/_temp/-2070376416/cmake-3.23.2-macos-universal/CMake.app/Contents/bin/cmake -E make_directory /Users/runner/builddir/output/bin/qfield.app/Contents/share/gdal
2022-09-26T04:10:41.0764810Z /Users/runner/work/_temp/-2070376416/cmake-3.23.2-macos-universal/CMake.app/Contents/bin/cmake -E copy_directory /Users/runner/builddir/vcpkg_installed/x64-osx/share/gdal /Users/runner/builddir/output/bin/qfield.app/Contents/share/gdal
2022-09-26T04:10:41.3238480Z /Users/runner/work/_temp/-2070376416/cmake-3.23.2-macos-universal/CMake.app/Contents/bin/cmake -E make_directory /Users/runner/builddir/output/bin/qfield.app/Contents/share/qfield
2022-09-26T04:10:41.3354220Z /Users/runner/work/_temp/-2070376416/cmake-3.23.2-macos-universal/CMake.app/Contents/bin/cmake -E copy_directory /Users/runner/work/QField/QField/resources/ /Users/runner/builddir/output/bin/qfield.app/Contents/share/qfield
```

As you can see, the files are copied into the `/Users/runner/builddir/output/bin/qfield.app/Contents/share/` folder. That's  where it should go for the main qfield.app, however it also means that the qfield_spix executable just fails to locate these as it expects the files to be relative to the executable (../share/). 

Two-part solution implemented here:
- when a PROJ_LIB environment variable is set, respect that (this is a no-brainer, I think it's a good behavior to adopt no matter what)
- when a _new_ QGIS_PKG_DATA_PATH environment variable is set, use that string when calling QgsApplication::setPkgDataPath (this is a bit more controversial a thing I'd guess, open to discussing)

**Comment by nirvn** [comment]
Actually I have a better idea here, that'll do a better job a testing an environment that matches our shipped app.

**Comment by nirvn** [comment]
Woupidou:
image

File: src/core/platforms/platformutilities.cpp [hunk 1]
```diff
@@ -107,7 +107,8 @@ QString PlatformUtilities::systemSharedDataLocation() const
   const static QString sharePath = QDir( QFileInfo( !QCoreApplication::applicationFilePath().isEmpty() ? QCoreApplication::applicationFilePath() : QCoreApplication::arguments().value( 0 ) ).canonicalPath()
                                          + QLatin1String( "/../share" ) )
                                      .absolutePath();
-  return sharePath;
+  const static QString environmentSharePath = QDir( QString( qgetenv( "QFIELD_SYSTEM_SHARED_DATA_PATH" ) ) ).absolutePath();
```
Comment (m-kuhn) [inline src/core/platforms/platformutilities.cpp:110]: Can you describe the order of precedence / influence of env/cmake etc. variables in the docstring and/or header inside this method?

**Comment by m-kuhn** [comment]
LGTM, except for formalities
thanks

File: src/core/platforms/platformutilities.cpp (hunk 1 above)
Comment (nirvn) [reply src/core/platforms/platformutilities.cpp:115]: @m-kuhn , good point, done.

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**